capa/ida/plugin/form.py

```diff
@@ -160,7 +160,8 @@ def __init__(self, name: str, option=Options.DEFAULT):
 
         # caches used to speed up capa explorer analysis - these must be init to None
         self.resdoc_cache: Optional[capa.render.result_document.ResultDocument] = None
-        self.ruleset_cache: Optional[capa.rules.RuleSet] = None
+        self.program_analysis_ruleset_cache: Optional[capa.rules.RuleSet] = None
+        self.rulegen_ruleset_cache: Optional[capa.rules.RuleSet] = None
```

Comment:
Adding second ruleset cache for the rule generator. This avoids multiple bugs where using a single ruleset cache may invalidate it in one view but not the other e.g. clicking `Reset` in rule generator clears the cache which causes the program analysis view to throw exceptions if the user does not re-analyze before using the UI.

@mr-tz we should be sure to pull these changes into #1292 before merging that PR.